openpype/hosts/blender/api/plugin.py

```diff
@@ -40,9 +43,16 @@ def get_unique_number(
     avalon_container = bpy.data.collections.get(AVALON_CONTAINERS)
     if not avalon_container:
         return "01"
-    asset_groups = avalon_container.all_objects
-
-    container_names = [c.name for c in asset_groups if c.type == 'EMPTY']
+    # Check the names of both object and collection containers
+    obj_asset_groups = avalon_container.all_objects
+    obj_group_names = [
+        c.name for c in obj_asset_groups
+        if c.type == 'EMPTY' and c.get(AVALON_PROPERTY)]
+    coll_asset_groups = avalon_container.children_recursive
+    coll_group_names = [
+        c.name for c in coll_asset_groups
+        if c.get(AVALON_PROPERTY)]
+    container_names = obj_group_names + coll_group_names
```

Comment:
Since this is being used as a lookup these should really be sets.

```suggestion
    # Check the names of both object and collection containers
    obj_asset_groups = avalon_container.all_objects
    obj_group_names = {
        c.name for c in obj_asset_groups
        if c.type == 'EMPTY' and c.get(AVALON_PROPERTY)}
    coll_asset_groups = avalon_container.children_recursive
    coll_group_names = {
        c.name for c in coll_asset_groups
        if c.get(AVALON_PROPERTY)}
    container_names = obj_group_names + coll_group_names
```

Also depending on the amount of containers I definitely wonder whether it's not faster to do the individual name checks instead of prepopulating just do:

```python
def has_child(obj, name):
    if obj.all_objects.get(name):
        return True
    if obj.children.get(name):
        return True

name = f"{asset}_{count:0>2}_{subset}"
while has_child(avalon_container, name):
    count += 1
    name = f"{asset}_{count:0>2}_{subset}"
```
_I don't use blender, so no idea if this would work_.

Also, what's the reason we care about recursive children for the Collections? What are we expecting in there?

Reply:
Mmmh I think that layouts, at the time when this was implemented, kept the name of the children without changing them. Now it isn't like that anymore, because any group always append the name of the group before the name of the object 🤔 I think we can remove it!

Btw, it doesn't matter anymore, but `obj.children` actually looks every time among all objects if the object is set as the parent, and doesn't store the children in the object 😅 ( https://docs.blender.org/api/3.1/bpy.types.Object.html#bpy.types.Object.children )
That's why I prepopulate the lists!

Reply:
Unfortunately I don't understand your last comment and what that means. But it sounds like you're going to try some things and tweak it so let me know when you want me to review that.

Reply:
Sorry 😅 but done!

Reply:
No worries, code looks much better now.